### PR TITLE
8300769: Remove G1CollectionSet::_inc_bytes_used_before

### PR DESCRIPTION
**Pull request by tschatzl**
Hi all,

  please review this removal of `G1CollectionSet::_inc_bytes_used_before`; this variable holds the amount of bytes used in eden regions, updated during mutator time. At the start of gc its value is copied to `G1CollectionSet::_bytes_used_before`, and then never touched again. The `_bytes_used_before` is read at end of gc, and then reset.

This change simply removes `_inc_bytes_used_before` and directly uses `_bytes_used_before` to accumulate eden region used bytes.

Testing: gha

Thanks,
  Thomas

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8300769](https://bugs.openjdk.org/browse/JDK-8300769): Remove G1CollectionSet::_inc_bytes_used_before

### Reviewers
 * [Albert Mingkun Yang](https://openjdk.org/census#ayang) (@albertnetymk - **Reviewer**) ⚠️ Review applies to [cbf4de63](https://git.openjdk.org/jdk/pull/12113/files/cbf4de63a348765d198ac5b8014c6a8851c02b5a)
 * [Ivan Walulya](https://openjdk.org/census#iwalulya) (@walulyai - **Reviewer**) ⚠️ Review applies to [cbf4de63](https://git.openjdk.org/jdk/pull/12113/files/cbf4de63a348765d198ac5b8014c6a8851c02b5a)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12113/head:pull/12113` \
`$ git checkout pull/12113`

Update a local copy of the PR: \
`$ git checkout pull/12113` \
`$ git pull https://git.openjdk.org/jdk pull/12113/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12113`

View PR using the GUI difftool: \
`$ git pr show -t 12113`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12113.diff">https://git.openjdk.org/jdk/pull/12113.diff</a>

</details>

### CODE REVIEW

**Review by albertnetymk — APPROVED** [review]
I think `_bytes_used` is enough; the `_before` suffix adds nothing and it's confusing -- there's no counterpart like `_bytes_used_after`.

**Comment by tschatzl** [comment]
> I think `_bytes_used` is enough; the `_before` suffix adds nothing and it's confusing -- there's no counterpart like `_bytes_used_after`.

I considered that, but you could argue that the metric gathered here is "bytes used before" (gc, in the collection set). The "bytes used after" is calculated on the spot when it's needed just before passing it on to JFR.
I'm good either way, let's wait on the second reviewer - if he recommends to change the name, I'll do the rename (in another RFE).

**Review by walulyai — APPROVED** [review]
Lgtm!

**Comment by tschatzl** [comment]
@albertnetymk found that actually the information of `_byte_used_before` can be recreated from `FreeCSetStats`, so `G1CollectionSet::_bytes_used_before` can be removed as well.
Further I fixed the merge errors with latest version.

File: src/hotspot/share/gc/g1/g1YoungGCPostEvacuateTasks.cpp [hunk 1]
```diff
@@ -465,6 +465,7 @@ class FreeCSetStats {
 
   void report(G1CollectedHeap* g1h, G1EvacInfo* evacuation_info) {
     evacuation_info->set_regions_freed(_regions_freed);
+    evacuation_info->set_collection_set_used_before(_after_used_bytes + _before_used_bytes);
```
Comment (albertnetymk) [inline src/hotspot/share/gc/g1/g1YoungGCPostEvacuateTasks.cpp:468]: The ordering, `before + after`, is more natural, IMO.

**Review by albertnetymk — APPROVED** [review]

**Review by walulyai — APPROVED** [review]
Better!

File: src/hotspot/share/gc/g1/g1YoungGCPostEvacuateTasks.cpp (hunk 1 above)
Comment (tschatzl) [reply src/hotspot/share/gc/g1/g1YoungGCPostEvacuateTasks.cpp]: Fixed

**Comment by tschatzl** [comment]
Thanks for your reviews
/integrate

**Pull request closed without merging**